Readme.md

```diff
@@ -237,6 +237,85 @@ Your attributes will be automatically saved when you end the session, but if the
 
 We have wrapped up the above example into a high/low number guessing game skill you can [download here](https://github.com/alexa/skill-sample-nodejs-highlowgame).
 
+### Additional Event Processing
+
+The `OnRequest` event is similar to the `NewSession` event but is called regardless of the value of `event.session.new`. If you define `OnRequest` you cannot define `NewSession`. In fact all events will go to `OnRequest` and you will need to call another emit to have processing continue:
```

Comment:
It should be state specific. I might have OnRequest defined for one state but it might not make sense for another state. I could call emit/emitWithState to call OnRequest in the main (or another) state if the need arose. 

I am working on a game that has a SetupState and GameState among others. The GameState uses OnRequest as it needs to call out to a PlayFab API, but the other states don't implement it.  

If it is a new session, then I call playFabService.Login which returns a result with some current state stored in PlayFab's data store. If it is not a new session, then I call playFabService.GetPlayerData to get the most recent data for the player. Both Login and GetPlayerData return the same data and I want to call the API before each request/intent during the same Alexa skill session to make sure I always have the most recent data in my skill from the PlayFab data store. I then call :foward to continue processing to the originally-determined intent.

I added this.data because you can't return values from emitted events. I had a value in OnRequest that I wanted to use in a subsequent event and I didn't want to pass it as a parameter to every emitWithState call to get to the point where it was used. I didn't need to persist it (either as request/response state or in DynamoDB) so this.attributes was the wrong place to put it.